src/psyclone/psyir/transformations/read_only_verify_trans.py

```diff
@@ -0,0 +1,138 @@
+        derived class
+
+    '''
+    from psyclone.psyir import nodes
+    from psyclone import psyGen
+    # The types of node that this transformation can enclose
+    valid_node_types = (nodes.Loop, psyGen.Kern, psyGen.BuiltIn,
+                        psyGen.Directive, nodes.Literal, nodes.Reference)
+
+    def __init__(self, node_class=ReadOnlyVerifyNode):
+        super(ReadOnlyVerifyTrans, self).__init__(node_class=node_class)
+
+    def __str__(self):
+        return ("Create a sub-tree of the PSyIR that has a "
+                "ReadOnlyVerifyNode at its root.")
+
+    @property
+    def name(self):
+        ''' Returns the name of this transformation as a string.'''
+        return "ReadOnlyVerifyTrans"
+
+    def validate(self, node_list, options=None):
+        '''Performs validation checks specific to read-only-based
+        transformations.
+
+        :param node_list: the list of Node(s) we are checking.
+        :type node_list: list of :py:class:`psyclone.psyir.nodes.Node`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :raises TransformationError: if distributed memory is configured.
+        :raises TransformationError: if transformation is applied to a \
+                                     Kernel or a BuiltIn call without its \
+                                     parent Loop.
+        :raises TransformationError: if transformation is applied to a Loop \
+                                     without its parent Directive when \
+                                     optimisations are applied.
+        :raises TransformationError: if transformation is applied to an \
+                                     orphaned Directive without its parent \
+                                     Directive.
+        '''
+
+        # Check ReadOnlyVerifyTrans specific constraints.
+
+        # Check constraints not covered by valid_node_types for
+        # individual Nodes in node_list.
+        from psyclone.psyir.nodes import Loop
+        from psyclone.psyGen import Directive, \
+            OMPParallelDirective, ACCParallelDirective
+
+        for node in node_list:
+
+            # Check that a ReadOnlyVerifyNode is not inserted between a Loop
+            # and its parent Directive when optimisations are applied, as
+            # this may result in including the end Directive for verification
+            # but not the beginning.
+            if isinstance(node, Loop) and isinstance(node.parent, Schedule) \
+               and isinstance(node.parent.parent, Directive):
+                raise TransformationError(
+                    "Error in {0}: Extraction of a Loop without its parent "
+                    "Directive is not allowed.".format(str(self.name)))
+
+            # Check that the ReadOnlyVerifyNode is not inserted within a thread
+            # parallel region when optimisations are applied. For instance,
```

Comment:
I think the first sentence in this comment is sufficient.

Reply:
Done.